homeassistant/components/device_tracker/config_entry.py

```diff
@@ -112,3 +123,34 @@ def state_attributes(self):
             attr[ATTR_BATTERY_LEVEL] = self.battery_level
 
         return attr
+
+
+class NetworkDeviceTrackerEntity(Entity):
+    """Represent a tracked network device."""
+
+    _remove_timer = None
+
+    @property
+    def state(self) -> str:
+        """Return the state of the device."""
+        @callback
+        def _update(_):
+            """Timer callback for device away."""
+            self.async_schedule_update_ha_state()
+            self._remove_timer = None
+
+        if self._remove_timer is not None:
+            self._remove_timer()
+            self._remove_timer = None
+
+        if dt_util.utcnow() - self.last_seen < DEFAULT_CONSIDER_HOME:
```

Comment:
properties should never have side effects. Instead, just add a new method on the entity `async_see`, that sets the state and starts the timer.

Reply:
Allright makes sense. How should we trigger async_see? We have state and last_seen as properties.

Reply:
Like this: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/owntracks/device_tracker.py#L25-L37

Reply:
OK, so each component register their own async see implementation?

Reply:
I thought we would put the complex logic in the person component and just have device trackers report state as is?

Reply:
Will I mightve misunderstood @balloob but he wanted a copy of the device from legacy implementation

Reply:
This is fixed b dmulcaheys PR